src/backends/tensorrt/plan_backend.cc

```diff
@@ -3161,13 +3206,26 @@ PlanBackend::Context::Run(
             "failed to run TRT response");
       }
 
-      // Process the output tensors with the device memory address even if
-      // zero-copy may be used, so that the memory copies perform asynchronously
-      // and wait for model execution.
-      payload_->responder_->ProcessTensor(
-          name, dt, batchn_shape,
-          static_cast<const char*>(io_binding_info.device_buffer_),
-          TRITONSERVER_MEMORY_GPU, gpu_device_);
+      // Use pinned memory address if zero copy is supported, else use device
+      // memory address.
+      if (zero_copy_support_) {
+        // std::string name_str = name;
+        LOG_VERBOSE(1) << "cudaLaunchHostFunc ProcessTensor";
+        cudaStreamWaitEvent(stream_, events_[next_set_].ready_for_output_, 0);
+        processTensor_tuples_.push_back(
+            {name, dt, batchn_shape,
+             static_cast<const char*>(io_binding_info.buffer_),
+             io_binding_info.memory_type_, io_binding_info.memory_type_id_,
+             payload_->responder_.get()});
+        cudaLaunchHostFunc(
+            stream_, ProcessTensorCudaHost,
+            reinterpret_cast<void*>(&processTensor_tuples_.back()));
```

Comment:
I would not pass the address of an element of vector to this function. The cudaLaunchHostFunc will be executed at a later point when ready_for_output_ event is recorded. The  addresses of processTensor_tuples_ are likely to get invalidated with growing tuples.